kernel/src/main/java/org/kframework/kbmc/KBMCOptions.java

```diff
@@ -0,0 +1,58 @@
+package org.kframework.kbmc;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParametersDelegate;
+import org.kframework.main.GlobalOptions;
+import org.kframework.unparser.PrintOptions;
+import org.kframework.utils.errorsystem.KEMException;
+import org.kframework.utils.file.FileUtil;
+import org.kframework.utils.inject.RequestScoped;
+import org.kframework.utils.options.DefinitionLoadingOptions;
+import org.kframework.utils.options.SMTOptions;
+
+import java.io.File;
+
+@RequestScoped
+public class KBMCOptions {
+
+    @ParametersDelegate
+    public transient GlobalOptions global = new GlobalOptions();
+
+    @ParametersDelegate
+    public DefinitionLoadingOptions definitionLoading = new DefinitionLoadingOptions();
+
+    @Parameter(names={"--spec-file"}, description="Path to the specification file.")
+    public String filePath;
+
+    private File specFile;
+
+    public synchronized File specFile(FileUtil files) {
+        if (specFile == null) {
+            if (filePath == null || filePath.equals("")) {
+                throw KEMException.criticalError("You have to provide exactly one main file in order to do outer parsing.");
+            }
+            specFile = files.resolveWorkingDirectory(filePath);
+        }
+        return specFile;
+    }
+
+    @ParametersDelegate
+    public SMTOptions smt = new SMTOptions();
+
+    @ParametersDelegate
+    public PrintOptions print = new PrintOptions();
+
+    @Parameter(names={"--spec-module", "-sm"}, description="Name of module containing specification to model check.")
+    public String specModule;
+
+    @Parameter(names={"--def-module", "-m"}, description="Name of module containing definition to model check under.")
+    public String defModule;
+
+    @Parameter(names="--depth", description="The maximum number of computational steps to model check.")
+    public Integer depth;
```

Comment:
you seem to have forgotten to pass this anywhere.

Reply:
done